models/user/pin.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package user
+
+import (
+	"encoding/json"
+)
+
+func GetPinnedRepositoryIDs(userID int64) ([]int64, error) {
+	pinnedstring, err := GetUserSetting(userID, PinnedRepositories)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(pinnedstring) == 0 {
+		return nil, nil
+	}
+	var parsedValues []int64
```

Comment:
```suggestion
	var parsedValues []int64
	if pinnedstring == "" {
		return parsedValues, nil
	}
```
Should make all subsequent calls far easier and less error-prone to handle, I'd say.

Reply:
I believe these are equivalent...?

Reply:
Actually not.
`var parsedValues []int64` initializes an empty list, `nil` of course doesn't.
As you are right now accessing it always using `range`, it doesn't make a difference because that handles nil correctly, but if you were to use the resulting list directly, you could get an unintended NPE.
Regarding the string comparison:
Yes that's equivalent, I simply find the suggested alternative easier to read.